inst/plumber/v1/plumber.R

```diff
@@ -1,312 +1,41 @@
-  params <- req$argsQuery
-  params$lkup <- lkups
-  do.call(pipapi::ui_pc_charts, params)
-}
-
-#* Return data for Poverty Calculator download
-#* @get /api/v1/pc-download
-#* @param country:[chr] Country ISO3 code
-#* @param year:[chr] Year
-#* @param povline:[dbl] Poverty Line
-#* @param fill_gaps:[bool] Fill gaps for years with no available surveys
-#* @param aggregate:[bool] Whether to aggregate results or not
-#* @param group_by:[chr] Triggers sub-groups aggregation
-#* @param welfare_type:[chr] Welfare Type. Options are "income" or "consumption"
-#* @param reporting_level:[chr] Reporting level. Options are "all", national", "urban", "rural".
-#* @serializer csv
-function(req) {
-  params <- req$argsQuery
-  params$lkup <- lkups
-  params$pop_units <- 1
-  do.call(pipapi::ui_pc_charts, params)
-}
-
-#* Return regional aggregations for all years
-#* @get /api/v1/pc-regional-aggregates
-#* @param povline:[dbl] Poverty Line
-#* @serializer json
-function(req) {
-  params <- req$argsQuery
-  params$lkup <- lkups
-  do.call(pipapi::ui_pc_regional, params)
-}
 
-# UI Endpoints: Country Profiles ----------------------------------------
-
-#* Return Country Profile - Key Indicators
-#* @get /api/v1/cp-key-indicators
-#* @param country:[chr] Country ISO3 code
-#* @param povline:[dbl] Poverty Line
-#* @serializer json
-function(req) {
-  params <- req$argsQuery
-  params$lkup <- lkups
-  do.call(pipapi::ui_cp_key_indicators, params)
-}
-
-
-#* Return Country Profile - Charts
-#* @get /api/v1/cp-charts
-#* @param country:[chr] Country ISO3 code
-#* @param povline:[dbl] Poverty Line
-#* @serializer json
-function(req) {
-  params <- req$argsQuery
-  params$lkup <- lkups
-  do.call(pipapi::ui_cp_charts, params)
-}
-
-# UI Endpoints: Survey metadata  ------------------------------------------
-
-#* Return data for the Data Sources page
-#* @get /api/v1/survey-metadata
-#* @param country:[chr] Country ISO3 code
-#* @serializer json list(na="null")
-function(req) {
-  params <- req$argsQuery
-  params$lkup <- lkups
-  do.call(pipapi::ui_svy_meta, params)
-}
+plumber::pr("inst/plumber/v1/endpoints.R") %>%
+  # pre-route log
+  plumber::pr_hook("preroute", function() {
+    tictoc::tic() # Start timer for log info
+  }) %>%
+  # post-route log
+  plumber::pr_hook("postroute", function(req, res) {
+    end_route <- tictoc::toc(quiet = TRUE)
+    log_info('{convert_empty(req$REMOTE_ADDR)} {convert_empty(req$REQUEST_METHOD)} {convert_empty(req$PATH_INFO)} {convert_empty(req$QUERY_STRING)}  {convert_empty(res$status)} {round(end_route$toc - end_route$tic, digits = getOption("digits", 6))}')
+  }) %>%
+  # pre-serialization log
+  plumber::pr_hook("preserialize", function() {
+    tictoc::tic()
+  }) %>%
+  # post-serialization log
+  plumber::pr_hook("preserialize", function(req) {
+    end_serial <- tictoc::toc(quiet = TRUE)
+    log_info('{convert_empty(req$PATH_INFO)} {round(end_serial$toc - end_serial$tic, digits = getOption("digits", 6))}') }) %>%
```

Comment:
Wondering if we should add a message/statement that this new line printed in the log is the serialization time for the previous request. It is not immediately clear from viewing the logs. 

```
INFO [2021-09-09 08:53:06] 127.0.0.1 GET /api/v1/pip ?country=all&year=all  200 38.81
INFO [2021-09-09 08:53:06] /api/v1/pip 0.06
```

Reply:
I had it initially but then thought it would be a lot of redundant information. We just need to explain it once to ITS. The final log will likely look different anyway (limited to pre- post-route)

Reply:
Gotcha. Yeah, then I agree. Let's keep it as is for now.